### ISSUE TITLE
How Can I Use & Share My Add-In?

### ISSUE THREAD

**Issue by IAmTheNewGuy** [OfficeDev/office-js-docs-pr#2943]
I have been building a taskpane add-in for Word using the Office Yeoman Generator. After following the directions found at <https://docs.microsoft.com/en-us/office/dev/add-ins/publish/publish-add-in-vs-code>, I can't access my add-in anywhere other than on the device I have used to make it. I don't know what files need to be shared in order to run an add-in. I am making an add-in for a teacher (who doesn't have Git or Node.js) and I would like to know how to get it to them, and them alone. I think it may also be useful to know I am not working for a company or part of an organization, and I do not have SharePoint. I've been searching for weeks, all over the Office Add-ins documentation, but I haven't been able to find (or maybe just understand) how to get this add-in to just one person. This is my first add-in, and has also been my first project with JavaScript. Should I get SharePoint just for this add-in, or where else could I go to find out how to share this add-in with just the teacher?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0b9105a4-a164-1e91-11ad-8ccb4f4d6b2d
* Version Independent ID: f4e5fae7-ac16-f852-8529-6b0fdd7bdad4
* Content: [Publish an add-in using Visual Studio Code and Azure - Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/publish/publish-add-in-vs-code)
* Content Source: [docs/publish/publish-add-in-vs-code.md](https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/publish/publish-add-in-vs-code.md)
* Product: **non-product-specific**
* Technology: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by AlexJerabek** [comment]
Hi @IAmTheNewGuy,

I'm happy you're making an add-in to help an educator. The only file the teacher will need is the manifest. As long as the web service powering your add-in is running and the manifest points to that web service,  anyone with that manifest can run your add-in. In these one-off scenarios without SharePoint or a larger M365 license with admin controls, you can have the person sideload the add-in once and it should stay loaded unless they clear the Office cache. In the [Deploy and publish Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/publish/publish) article, you would be looking at the "Sideloading" and "Network share" options for these small-scale solutions.

Let me tag @davidchesnut for any additional guidance here. He's set up a lot of our samples to use GitHub hosting, which may be a solution to your side of the web service hosting.

**Comment by davidchesnut** [comment]
Hi @IAmTheNewGuy,

Real quick I want to point out that if you are doing something relatively simple, you may find [Office Scripts](https://docs.microsoft.com/en-us/office/dev/scripts/) easier to work with. You can share them through OneDrive without the need to configure your own web server.

If you want to try GitHub hosting you can see one of our samples using it: [Excel keyboard shortcuts](https://github.com/OfficeDev/PnP-OfficeAddins/tree/main/Samples/excel-keyboard-shortcuts). The steps walk you through how to sideload the manifest. All the HTML and web files are hosted from the GitHub folder location itself.

The key parts are:
1. You have to set up GitHub pages on one of your repos. Just configure the pages to use the root of your repo.
2. Copy the web pages for the add-in into a folder on your repo (similar to how Excel keyboard shortcuts is in a folder.)
3. Update the manifest to point to the GitHub pages versions of the HTML. See [Excel keyboard shortcuts manifest](https://github.com/OfficeDev/PnP-OfficeAddins/blob/main/Samples/excel-keyboard-shortcuts/manifest.xml) for an example of how we configured the URLs for GitHub hosting.
4. Whoever wants to use your add-in just needs to sideload, or use a network share, the manifest.

Also just want to point out that using GitHub only works for static web pages. And everything will be publicly available.

Cheers,
David

**Comment by davidchesnut** [comment]
Hi @IAmTheNewGuy,

Also wanted to mention regarding the [Publish an add-in developed with Visual Studio Code](https://docs.microsoft.com/office/dev/add-ins/publish/publish-add-in-vs-code) article. I'd try opening the task pane HTML page directly in a browser. All of the files (HTML, JS, etc...) should be available through a browser by using a URL to access them. If this works on one computer, but not another, you should get an error that will help to start to debug what's going on. 

Cheers,
David

**Comment by IAmTheNewGuy** [comment]
Hey, thank you guys so much! I found out how to use network share to get the add-in to work on another device. I also realized that when I read "change any reference to your localhost URL (such as https://localhost:3000) to the URL you've copied," on <https://docs.microsoft.com/en-us/office/dev/add-ins/publish/publish-add-in-vs-code>, I ended up replacing the app-domain, thinking it also needed to be changed. After a quick fix, it works great! Once I improve my work a little more, it may be publicly available on GitHub.
